model-optimizer/extensions/middle/L2NormToNorm.py

```diff
@@ -82,6 +82,27 @@ def replace_pattern(self, graph: Graph, match: dict):
             log.debug('The value of the "maximum_y_data" is not defined or is not constant')
             return
 
+        # We need to check axes which performed reduction because IE supports only 2D, 3D, 4D inputs and
+        # reduction only along spatial dimensions.
+        input_shape = match['sum'].in_port(0).data.get_shape().size
+        if input_shape not in [2, 3, 4]:
+            log.debug('IE supports L2 normalization only for 2D, 3D and 4D tensors, skip fusing transformation.')
+            return
+
+        axes = match['sum'].in_port(1).data.get_value()
+        axes = int64_array(axes)
+        if axes.shape == ():
+            axes = int64_array([axes])
+        axes.sort()
+
+        across_spatial = False
+        if np.array_equal(axes, int64_array(np.arange(start=1, stop=input_shape))):
```

Comment:
`1` is not spatial dimension.

Reply:
It is middle stage and this transformation is specific for TensorFlow only, so this operation in `NHWC` layout.

Reply:
Please make a comment about this. Then you need to check layout and I think the correct reference operand `int64_array(np.arange(start=1, stop=input_shape-1))` where you also need to exclude `C` channel.

Reply:
>It is middle stage and this transformation is specific for TensorFlow only, so this operation in NHWC layout.

TensorFlow models can be in NCHW layout. So the layout should be explicitly checked.

Reply:
across_spatial is a flag that specifies if normalization is performed over CHW or HW. Maybe be better if I rename this variable, but I am still think that logic is right.

Reply:
what is purpose of  `channel_shared` ?

Reply:
`channel_shared` is a flag that specifies if scale parameters are shared across channels.
It is Caffe-specific attribute and looks like it doesn't applicable to this fused operation.

Reply:
I got it about `channel_shared`, thanks. Regarding `across_spatial` I think you need to make a comment that it means normalization excepting batch dimension only. Otherwise, it creates confusion since channel dimension is not a spatial dimension.

Reply:
Transformation was refactored and this code was removed. In new implementation comments were updated.